### PR TITLE
Sql store for redirects

### PR DESCRIPTION
**Pull request by marisks**
The main changes:
- SQL store instead of DDS
- Migration from DDS to SQL

### CODE REVIEW

File: src/Controllers/NotFoundRedirectController.cs [hunk 1]
```diff
@@ -24,11 +23,21 @@ namespace BVNetwork.NotFound.Controllers
...
+            DdsRedirectRepository ddsRedirectRepository)
+        {
+            _redirectsService = redirectsService ?? throw new ArgumentNullException(nameof(redirectsService));
+            _ddsRedirectRepository = ddsRedirectRepository ?? throw new ArgumentNullException(nameof(ddsRedirectRepository));
+        }
+
         private void CheckAccess()
         {
-            if (!PrincipalInfo.HasEditAccess)
+            if (!EPiServer.Security.PrincipalInfo.HasEditAccess)
```
Comment (gatisb) [inline src/Controllers/NotFoundRedirectController.cs]: do we need full namespace here?

Comment (marisks) [reply src/Controllers/NotFoundRedirectController.cs]: Yes, we need. Episerver.Security namespace contains some extensions which break Linq in other places.

File: src/Core/CustomRedirects/DefaultRedirectsService.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        public IEnumerable<CustomRedirect> GetAll()
+        {
+            return _redirectLoader.GetAll();
+        }
+
+        public IEnumerable<CustomRedirect> GetAllExcludingIgnored()
+        {
+            return _redirectLoader.GetByState(RedirectState.Saved);
```
Comment (gatisb) [inline src/Core/CustomRedirects/DefaultRedirectsService.cs]: this does not include suggestion

Comment (gatisb) [reply src/Core/CustomRedirects/DefaultRedirectsService.cs]: so maybe better name is needed

File: src/Core/Data/SqlRedirectRepository.cs [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+using System.Data.Common;
+using BVNetwork.NotFound.Core.CustomRedirects;
+using EPiServer.Data;
+using EPiServer.Logging;
+
+namespace BVNetwork.NotFound.Core.Data
+{
+    public class SqlRedirectRepository : IRepository<CustomRedirect>, IRedirectLoader
+    {
+        private static readonly ILogger Logger = LogManager.GetLogger();
```
Comment (gatisb) [inline src/Core/Data/SqlRedirectRepository.cs]: cant this be passed in trough constructor?

File: src/Core/CustomRedirects/DefaultRedirectsService.cs (hunk 2 above)
Comment (marisks) [reply src/Core/CustomRedirects/DefaultRedirectsService.cs]: Yes, this is only for redirects, not for suggestions.

File: src/Core/Data/SqlRedirectRepository.cs (hunk 3 above)
Comment (gatisb) [reply src/Core/Data/SqlRedirectRepository.cs]: does it need to be static?

Comment (marisks) [reply src/Core/Data/SqlRedirectRepository.cs]: Episerver's convention is this as I know. Haven't seen injected logger.

File: src/Core/Data/SqlRedirectRepository.cs [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            return ExecuteQuery(() =>
+                CreateCommand(
+                    sqlCommand,
+                    CreateIntParameter("state", (int)state)));
+        }
+
+        public IEnumerable<CustomRedirect> Find(string searchText)
+        {
+            var sqlCommand = $@"SELECT {AllFields} FROM {RedirectsTable}
+                                    WHERE OldUrl like '%' + @searchText + '%'
```
Comment (gatisb) [inline src/Core/Data/SqlRedirectRepository.cs]: Possible injection as not found url is controlled by end user.
Also sql repo should be protected against injection

Comment (marisks) [reply src/Core/Data/SqlRedirectRepository.cs]: Nope, this is not an injection. @searchText is a SQL parameter, not a string.

Comment (gatisb) [reply src/Core/Data/SqlRedirectRepository.cs]: ok

Comment (gatisb) [reply src/Core/Data/SqlRedirectRepository.cs]: you ar right not a problem both searchText variables having same name confused me

**Comment by tyng113** [comment]
Toyota_Motor_  Online_Promotion_Winning_Certificate

**Pull request merged**